target/portfolio-1/script.js

```diff
@@ -26,3 +26,33 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+function storeZipCode(){
+    event.preventDefault();
+    var zipcode = document.getElementById("zipcode").value;
+    localStorage.setItem("zipcode", zipcode);
+    window.location.href = "/repList.html";
+
+}
+function getRepList(){
+    var zipcode = localStorage.getItem("zipcode");
+    fetch(`/rep_list?zipcode=${zipcode}`).then(response => response.json()).then((representatives) => {
+        representatives = JSON.parse(representatives);
+        console.log(representatives);
+        var representativeList = document.getElementById("repList");
+        var offices = representatives.offices;
+        var officials = representatives.officials;
+        for (var i = 0; i < offices.length; i++) {
+            for (number of offices[i]["officialIndices"]){
+                console.log("i: " + i + " number: " + number + " array: " + offices[i]["officialIndices"]);
+                representativeList.appendChild(createListElement(offices[i]["name"] + ": " + officials[number]["name"]));
+            }
```

Comment:
I'm not entirely sure what is happening here. Are you list indexing with a string ? :O 
Is there some strange javascript thing that I don't know about yet

Reply:
the JSON object for offices contains an indices array for the corresponding officials in the officials array (i.e. indices = [3, 4, 5] means officials[3], officials[4], officials[5] are the people with that particular title)- hence the second loop